cvmfs/upload_local.cc

```diff
@@ -168,6 +167,9 @@ void LocalUploader::FinalizeStreamedUpload(UploadStreamHandle *handle,
               UploaderResults(UploaderResults::kChunkCommit, cpy_errno));
       return;
     }
+    if (!content_hash.HasSuffix()) {  // count only data, not metadata
+      CountUploadedBytes(GetFileSize(upstream_path_ + "/" + final_path));
```

Comment:
Sorry for looking only now: this might still not be enough.  Large files can be split in smaller pieces, in which case the hash has the 'P' suffix.